frontend-cert/js-projects/quiz-game/user-stories.md

```diff
@@ -1,15 +1,9 @@
-1. You should create an object named `questions` with three keys, `science`, `geography`, and `literature`, each with the value of an empty array.
-1. You should create at least four `question#` variables, where `#` is replaced by a number starting from `1`.
-1. Each `question#` variable should be an object with the keys `question`, `choices`, and `answer`.
+1. You should create an array named `questions`.
+1. The `questions` array should contain at least five objects, each having the keys `category`, `question`, `choices`, and `answer`.
+1. The `category` key should have the value of a string representing a question category.
 1. The `question` key should have the value of a string representing a question.
 1. The `choices` key should have the value of an array containing three strings, which are alternative answers to the question.
 1. The `answer` key should have the value of a string, representing the correct answer to the question. Also, the value of `answer` should be included in the `choices` array.
-1. You should push each `question#` to the respective category inside the `questions` object.
-1. Each category of the `questions` object should have at least one element.
-1. You should declare a variable named `category` which has either the value of `"science"`, `"geography"`, or `"literature"`.
-1. You should remove the first element from the `questions[category]` array and store it inside a variable named `chosenQuestion`.
-1. You should log the value of the `question` key of `chosenQuestion` to the console.
-1. You should log the value of the `choices` key of `chosenQuestion` to the console.
-1. You should declare a variable named `answer`, which is a string.
-1. You should create an `if` statement that checks if `answer` is the correct answer and logs the string `The answer is correct!` to the console.
-1. You should write an `else` statement that logs the string `Wrong answer, the correct answer is: <correct-answer>`, where `<correct-answer>` is the value of the correct answer to the chosen question.
\ No newline at end of file
+1. You should have a function named `getRandomQuestion` that returns a random question object from the `questions` array.
+1. You should have a function named `getRandomComputerChoice` that takes the array of the available choices as a parameter, and returns a random answer to the selected question.
+1. You should have a function named `getResults` that takes the selected question object as its parameter and returns `The computer's choice is correct!` if the answer is correct. Otherwise, it returns `The computer's choice is wrong. The correct answer is: <correct-answer>`, where `<correct-answer>` is the value of the correct answer to the chosen question.
```

Comment:
It is fine if you don't want to the multiple functions and just want one.

The main priority and goal is to get input from the "computer" and print the result.
So you can structure that however you like 👍

Reply:
What I was trying to say is that I think it's fine to ask them to write the functions without calling them and log the results.

So I would simply test that those three functions works correctly instead of checking this part:
```js
const questionObj = getRandomQuestion();
// campers can either choose to use destructuring or dot/bracket notation to get these variables
const { question, choices } = questionObj;

console.log(question);
console.log(`Choices: ${choices}`);

const computerChoice = getRandomComputerChoice(choices);
console.log(`Computer chooses: ${computerChoice}`);

const results = getResults(questionObj);
console.log(results);
```

Reply:
Yeah, that is fine